src/core/SkipToContent/SkipToContentMarker.tsx

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, useTheme } from '../utils';
+
+export type SkipToContentMarkerProps = {
+  /**
+   * Optional main content as child.
+   */
+  children?: React.ReactNode;
+} & Omit<CommonProps, 'id'>;
+
+/**
+ * Marks the start of the main content of the page.
+ * Used in conjunction with `SkipToContentLink`.
+ * @example <caption>`SkipToContentMarker` is placed before the main content</caption>
+ * <SkipToContentMarker /> <MainContentMarker />
+ * @example <caption>The main content is placed as a child of the `SkipToContentMarker` element</caption>
+ * <SkipToContentMarker> <MainContent /> </SkipToContentMarker>
+ */
+export const SkipToContentMarker = (props: SkipToContentMarkerProps) => {
```

Comment:
Looks like all this component does is attach an id, and it's not very flexible. So maybe it should be a hook that just gives you an id.

Also the id should probably not be hardcoded because someone might want multiple of these markers.

Usage could be something like this:

```tsx
const contentId = useSkipToContentMarker();
...
<SkipToContentLink href={`#${contentId}`}>Skip to content</SkipToContentLink>
```

But it's less useful now that React 18 offers a `useId` hook.

Reply:
I based this off of the example you suggested. I was going to do something more like this, but decided to do two components instead because our React stuff is component based and I didn't want to require users to fiddle with the ids on their components. I don't think adding an id to their components works as you suggest here - what if the component already has an id? I am also not sure how having multiple markers would work. What kind of situation would call for that?

Reply:
> I based this off of the example you suggested.

That example is not gospel, it was just the top one I found in my google search when creating that issue. You could do more research into other libraries to see how they do this.

> our React stuff is component based

This is no reason to limit any future implementations. If hooks can provide us more flexibility, then we shouldn't avoid it because we haven't done anything like it in the past.

> I don't think adding an id to their components works as you suggest here - what if the component already has an id?

Not sure what's the concern here. They are the ones adding the id (we are simply giving them the id). If there are multiple, then the last one will be used.

> I am also not sure how having multiple markers would work. What kind of situation would call for that?

Multiple markers also means multiple links. The ids need to be unique. I have a few examples in my other comment: https://github.com/iTwin/iTwinUI-react/pull/618#discussion_r844411643

Reply:
> Elements are supposed to only have one id.

That's true for HTML, but we are working in JSX, where props are just JS objects. Please see above where I said "the last one is used".

I'm still not sure what your concern here is. Nowhere am I suggesting having multiple ids on the same element. You can see a clearer example of what I'm suggesting in my other comment: https://github.com/iTwin/iTwinUI-react/pull/618#discussion_r849469581

Reply:
If you really want to keep this component, then it will need to be more flexible:
- it should allow overriding the id (for when there are multiple links/markers)
- it should be polymorphic to allow changing what element is rendered (`<div>` is not always desirable)
- it might need `tabindex=-1` for older browsers and for receiving focus if we progressively enhance the functionality using JS

Reply:
Removed the marker. Didn't add a hook because it seems unnecessary. Added a prop so that users are required to make it work.

Reply:
Why not just make `href` required?

Reply:
Didn't know you could do that

Reply:
Ok, I did that I think. I mean, it requires href now but this might not have been the method you were thinking of? The internet wasn't super helpful.